internal/action/generate.go

```diff
@@ -261,21 +261,40 @@ func clamp(mi, ma, value int) int {
 	return value
 }
 
-func (s *Action) generatePasswordForRule(ctx context.Context, c *cli.Context, length, name, domain string, rule pwrules.Rule) (string, error) {
+// generatePasswordForRule validates the user-provided password length against
+// the rule for the domain and condtionally prompts the user for a correct
+// length if the initial value is invalid.
+func (s *Action) generatePasswordForRule(ctx context.Context, length, domain string, rule pwrules.Rule) (string, error) {
 	out.Noticef(ctx, "Using password rules for %s ...", domain)
 
-	wl := 16
-	if iv, err := strconv.Atoi(length); err == nil {
-		wl = clamp(rule.Minlen, rule.Maxlen, iv)
-		debug.Log("restricting requested password length (%s) to %d because of the rule {%d,%d}", length, wl, rule.Minlen, rule.Maxlen)
-	}
+	var iv int
+	var err error
 
-	question := fmt.Sprintf("How long should the password be? (min: %d, max: %d)", rule.Minlen, rule.Maxlen)
-	iv, err := termio.AskForInt(ctx, question, wl)
-	if err != nil {
+	if iv, err = strconv.Atoi(length); err != nil {
 		return "", exit.Error(exit.Usage, err, "password length must be a number")
 	}
 
+	wl := clamp(rule.Minlen, rule.Maxlen, iv)
+	if wl != iv {
+		debug.Log(
+			"setting requested password length (%s) to %d because of the rule {min: %d, max: %d}",
+			length, wl, rule.Minlen, rule.Maxlen,
+		)
+
+		question := fmt.Sprintf(
+			"How long should the password be? (min: %d, max: %d)",
+			rule.Minlen, rule.Maxlen,
+		)
+
+		var sv string
+
+		if sv, err = termio.AskForString(ctx, question, strconv.Itoa(wl)); err != nil {
+			return "", err
+		}
+
+		return s.generatePasswordForRule(ctx, sv, domain, rule)
+	}
+
 	iv = clamp(rule.Minlen, rule.Maxlen, iv)
```

Comment:
I think this is never something else than `iv` itself given the above `if` conditions, no?

Reply:
you're right, good catch. this can be removed.